enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/state/CoreState.java

```diff
@@ -134,57 +134,34 @@ public synchronized void notifyCommitted( long commitIndex )
     private void submitApplyJob( long lastToApply )
     {
         applier.submit( ( status ) -> () -> {
-            try ( LogEntrySupplier cacheLogEntrySupplier = new CacheApplier();
-                    LogEntrySupplier cursorLogEntrySupplier = new CursorApplier() )
+            try ( LogEntrySupplier logEntrySupplier = new LogEntrySupplier() )
             {
-                LogEntrySupplier currentLogEntrySupplier = cacheLogEntrySupplier;
                 lastApplyingStorage.persistStoreData( lastToApply );
-                for ( long i = lastApplied + 1; !status.isCancelled() &&  i <= lastToApply; i++ )
+                for ( long logIndex = lastApplied + 1; !status.isCancelled() &&  logIndex <= lastToApply; logIndex++ )
```

Comment:
I disagree, having the condition in one place is a more readable solution than having `continue` or `break` in the `for` loop.